testdata/baselines/reference/submodule/conformance/elementAccessChain.3.js.diff

```diff
@@ -16,55 +16,31 @@
 -        }
 -    return t;
 -};
--(obj === null || obj === void 0 ? void 0 : obj["a"])++;
--(obj === null || obj === void 0 ? void 0 : obj.a["b"])++;
--(obj === null || obj === void 0 ? void 0 : obj["a"])--;
--(obj === null || obj === void 0 ? void 0 : obj.a["b"])--;
--++(obj === null || obj === void 0 ? void 0 : obj["a"]);
--++(obj === null || obj === void 0 ? void 0 : obj.a["b"]);
----(obj === null || obj === void 0 ? void 0 : obj["a"]);
----(obj === null || obj === void 0 ? void 0 : obj.a["b"]);
+ (obj === null || obj === void 0 ? void 0 : obj["a"])++;
+ (obj === null || obj === void 0 ? void 0 : obj.a["b"])++;
+ (obj === null || obj === void 0 ? void 0 : obj["a"])--;
+@@= skipped -20, +8 lines =@@
+ ++(obj === null || obj === void 0 ? void 0 : obj.a["b"]);
+ --(obj === null || obj === void 0 ? void 0 : obj["a"]);
+ --(obj === null || obj === void 0 ? void 0 : obj.a["b"]);
 -obj === null || obj === void 0 ? void 0 : obj["a"] = 1;
 -obj === null || obj === void 0 ? void 0 : obj.a["b"] = 1;
 -obj === null || obj === void 0 ? void 0 : obj["a"] += 1;
 -obj === null || obj === void 0 ? void 0 : obj.a["b"] += 1;
--for (obj === null || obj === void 0 ? void 0 : obj["a"] in {})
--    ;
--for (obj === null || obj === void 0 ? void 0 : obj.a["b"] in {})
--    ;
--for (obj === null || obj === void 0 ? void 0 : obj["a"] of [])
--    ;
--for (obj === null || obj === void 0 ? void 0 : obj.a["b"] of [])
--    ;
--({ a: obj === null || obj === void 0 ? void 0 : obj["a"] } = { a: 1 });
--({ a: obj === null || obj === void 0 ? void 0 : obj.a["b"] } = { a: 1 });
++(obj === null || obj === void 0 ? void 0 : obj["a"]) = 1;
```

Comment:
This seems to be incorrect, right? This is trying to assign 1 to the parenthesized expression?

Reply:
This is the structure we make in strada for these inputs, too, just without the parenthesis. The strada output does not roundtrip - the parens are actually required. And yes, they do make it clear that the output is... questionable. But that's fine, these all have a `The left-hand side of an assignment expression may not be an optional expression` error, which'd by why it's not handled sensibly in the transform - it's completely invalid.